I cannot find readline.h

### Output

```
checking for libedit readline replacement... yes
configure: error: Please reinstall libedit - I cannot find readline.h
```

### Expected Result

compiled php version

### Command

```
$ phpbrew --debug install 7.0 +everything
```

### Build Log

```
https://gist.github.com/oNdsen/f6231c0aef5a7492427a0689219fd518
```

### Platform

OS:
Debian GNU/Linux 8 (jessie)

Running PHP:
doesnt matter. tried with 5.6, 7.0, 7.1

Installing PHP:
tryed each one of the following versions:
7.1.7, 7.0.21, 7.0.20, 5.6.31

### Comment

this problem exists on 3 different debian servers. each server has allready compiled php versions installed (on january, via same phpbrew installation).
what ive done:

```
$ phpbrew update
$ phpbrew --debug install 7.0 +everything
```

so, readline is installed. i just cannot compile a new php version after i updated phpbrew (doesnt matter with or without extensions).

```
root@web01:~# apt-get install libedit2 libreadline6 libreadline6-dev
Reading package lists... Done
Building dependency tree
Reading state information... Done
libedit2 is already the newest version.
libreadline6 is already the newest version.
libreadline6-dev is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
```

```
root@web01:~# find / -name "readline.h"
/usr/include/readline/readline.h

```

--- a/src/PhpBrew/Patches/ReadlinePatch.php
+++ b/src/PhpBrew/Patches/ReadlinePatch.php
@@ -1,0 +1,97 @@
+<?php
+
+namespace PhpBrew\Patches;
+
+use CLIFramework\Logger;
+use PhpBrew\Buildable;
+use PhpBrew\PatchKit\DiffPatchRule;
+use PhpBrew\PatchKit\Patch;
+
+/**
+ * Fix support for libedit in PHP 5.4 to PHP 7.1.
+ */
+class ReadlinePatch extends Patch
+{
+    public function desc()
+    {
+        return 'fix readline detection on PHP 5.3 to PHP 7.1';
+    }
+
+    public function match(Buildable $build, Logger $logger)
+    {
+        return version_compare($build->getVersion(), '5.4', '>=')
+            || version_compare($build->getVersion(), '7.1', '<=');
+    }
+
+    /**
+     * @link https://git.php.net/?p=php-src.git;a=commit;h=1ea58b6e78355437b79fb7b1f287ba6688fb1c57
+     *
+     * @codeCoverageIgnore
+     */
+    public function rules()
+    {
+        return array(
+            DiffPatchRule::fromPatch(
+                <<<'EOP'
+From: =?UTF-8?q?Ond=C5=99ej=20Sur=C3=BD?= <ondrej@sury.org>
+Date: Mon, 17 Dec 2018 09:56:06 +0000
+Subject: Fix rl_completion_matches detection
+
+Also fix a typo when checking for rl_on_new_line in readline library.
+---
+ ext/readline/config.m4 | 17 +++++++++++++++--
+ 1 file changed, 15 insertions(+), 2 deletions(-)
+
+diff --git a/ext/readline/config.m4 b/ext/readline/config.m4
+index d63df8bef9..3995bd7587 100644
+--- a/ext/readline/config.m4
++++ b/ext/readline/config.m4
+@@ -60,13 +60,20 @@ if test "$PHP_READLINE" && test "$PHP_READLINE" != "no"; then
+     -L$READLINE_DIR/$PHP_LIBDIR $PHP_READLINE_LIBS
+   ])
+ 
+-  PHP_CHECK_LIBRARY(edit, rl_on_new_line,
++  PHP_CHECK_LIBRARY(readline, rl_on_new_line,
+   [
+     AC_DEFINE(HAVE_RL_ON_NEW_LINE, 1, [ ])
+   ],[],[
+     -L$READLINE_DIR/$PHP_LIBDIR $PHP_READLINE_LIBS
+   ])
+ 
++  PHP_CHECK_LIBRARY(readline, rl_completion_matches,
++  [
++    AC_DEFINE(HAVE_RL_COMPLETION_MATCHES, 1, [ ])
++  ],[],[
++    -L$READLINE_DIR/$PHP_LIBDIR $PHP_READLINE_LIBS
++  ])
++
+   AC_DEFINE(HAVE_LIBREADLINE, 1, [ ])
+ 
+ elif test "$PHP_LIBEDIT" != "no"; then
+@@ -114,11 +121,17 @@ elif test "$PHP_LIBEDIT" != "no"; then
+     -L$READLINE_DIR/$PHP_LIBDIR
+   ])
+ 
++  PHP_CHECK_LIBRARY(edit, rl_completion_matches,
++  [
++    AC_DEFINE(HAVE_RL_COMPLETION_MATCHES, 1, [ ])
++  ],[],[
++    -L$READLINE_DIR/$PHP_LIBDIR $PHP_READLINE_LIBS
++  ])
++
+   AC_DEFINE(HAVE_LIBEDIT, 1, [ ])
+ fi
+ 
+ if test "$PHP_READLINE" != "no" || test "$PHP_LIBEDIT" != "no"; then
+-  AC_CHECK_FUNCS([rl_completion_matches])
+   PHP_NEW_EXTENSION(readline, readline.c readline_cli.c, $ext_shared, cli)
+   PHP_SUBST(READLINE_SHARED_LIBADD)
+ fi
+-- 
+2.27.0
+
+EOP
+            )->strip(1)
+        );
+    }
+}

--- a/src/PhpBrew/Tasks/BeforeConfigureTask.php
+++ b/src/PhpBrew/Tasks/BeforeConfigureTask.php
@@ -6,6 +6,7 @@ use PhpBrew\Build;
 use PhpBrew\ConfigureParameters;
 use PhpBrew\Patches\Apache2ModuleNamePatch;
 use PhpBrew\Patches\FreeTypePatch;
+use PhpBrew\Patches\ReadlinePatch;
 use PhpBrew\PatchKit\Patch;
 
 /**
@@ -49,12 +50,14 @@ class BeforeConfigureTask extends BaseTask
             $this->logger->info('===> Checking patches...');
 
             $freeTypePatch = new FreeTypePatch();
-            $freeTypePatched = false;
+            $readlinePatch = new ReadlinePatch();
+            $needBuildConf = false;
 
             /** @var Patch[] $patches */
             $patches = array(
                 new Apache2ModuleNamePatch($build->getVersion()),
                 $freeTypePatch,
+                $readlinePatch
             );
 
             foreach ($patches as $patch) {
@@ -63,14 +66,14 @@ class BeforeConfigureTask extends BaseTask
                     $patched = $patch->apply($build, $this->logger);
                     $this->logger->info("$patched changes patched.");
 
-                    if ($patch === $freeTypePatch) {
-                        $freeTypePatched = $patched;
+                    if ($patch === $freeTypePatch || $patch === $readlinePatch) {
+                        $needBuildConf = $patched;
                     }
                 }
             }
 
-            if ($freeTypePatched) {
-                $this->logger->info('GD extension was patched, need to run buildconf');
+            if ($needBuildConf) {
+                $this->logger->info('Need to run buildconf');
 
                 $buildConf = new BuildConfTask($this->logger);
                 $buildConf->run($build);

